src/emulator/functionsEmulator.ts

```diff
@@ -169,7 +168,7 @@ export class FunctionsEmulator implements EmulatorInstance {
       const runtimeReq = http.request(
         {
           method,
-          path: "/" + trimFunctionPath(req.url),
+          path: req.url || "/",
```

Comment:
This change was just made (by me) to fix a different bug.  Aren't we un-fixing it now?

Reply:
False. Your fix allowed `req.path` to equal the last chunks of the path, dropping the `trigger_id`, `project_id` and `region`. In order to provide this information into the function runtime (and eventually `req.baseUrl` and `req.mountpoint`) we need to preserve this data and allow the runtime to do the split. The new system (implemented near `ephemeralApp`) uses a sub-router which automatically slices the path for us and sets `req.baseUrl` so we don't need this hack since the router takes care of it.

Reply:
Specifically that was fixed here:
https://github.com/firebase/firebase-tools/pull/1332

Reply:
OK you'll need to hand-hold me here.  What will they see if they run a function like this:
```js
exports.httpFn = functions.https.onRequest((request, response) => {
    response.send({
        method: request.method,
        path: request.path, 
        query: request.query 
    });
});
```

As of #1332 it matches prod, will that change?

Reply:
It will still match prod, as defined by [this test](https://github.com/firebase/firebase-tools/blob/master/src/test/emulators/functionsEmulator.spec.ts#L57). Please examine [`functionsEmulatorRuntime.ts#566`](https://github.com/firebase/firebase-tools/blob/ah-speedhax/src/emulator/functionsEmulatorRuntime.ts#L566) where we provide the new router. The chunks matched in the array passed to `use` will be dropped when the sub-router is called, the result should be the same as what you implemented, but now with `baseUrl` and `mountpoint` provided to the developer.

Reply:
Ok now I get it, thanks!